### PR TITLE
(feat) O3-3542 : Launch tutorials when clicking an item from the list

### PR DESCRIPTION
**Pull request by Vijaykv5**
## Requirements
- [x] This PR has a title that briefly describes the work done including a [conventional commit](https://o3-dev.docs.openmrs.org/#/getting_started/contributing?id=your-pr-title-should-indicate-the-type-of-change-it-is) type prefix and a Jira ticket number if applicable. See existing PR titles for inspiration.

#### For changes to apps
- [x]  My work conforms to the [**OpenMRS 3.0 Styleguide**](https://om.rs/styleguide) and [**design documentation**](https://zeroheight.com/23a080e38/p/880723-introduction).

#### If applicable
- [ ] My work includes tests or is validated by existing tests.

## Summary

- This PR adds a sample tutorial when clicking an tutorial from the tutorial modal.
- It makes use of `useContext` to manage states between multiple components and it's written in `onboarding-context` file.

## Screenshots

https://github.com/openmrs/openmrs-esm-user-onboarding/assets/94985341/a8277c3e-0d47-4172-8bd0-b4fe2d7f55ce

## Related Issue

[O3-3542 ](https://openmrs.atlassian.net/browse/O3-3542)

### CODE REVIEW

File: src/tutorial/modal.component.tsx [hunk 1]
```diff
@@ -2,12 +2,27 @@ import React from 'react';
...
+  const { startOnboarding } = useOnboarding();
+
+  const handleWalkthroughClick = (tutorial) => {
+    const steps = [
+      {
+        target: '[aria-label="OpenMRS"]',
+        content: 'This is the OpenMRS logo. Click here to go back to the home page.',
+        disableBeacon: true,
+      },
+    ];
```
Comment (Piumal1999) [inline src/tutorial/modal.component.tsx:22]: Let's move the steps to the config file itself. For now, we can just add one step for each walkthrough, displaying the name of the walkthrough.

**Review by Piumal1999 — COMMENTED** [review]
Good job @Vijaykv5! 
Requested a small change.

File: src/tutorial/modal.component.tsx (hunk 1 above)
Comment (Vijaykv5) [reply src/tutorial/modal.component.tsx:22]: @Piumal1999 
May be we can have like the content of the current walkthrough to the name of the walkthrough that's been clicked for now?
Or do we need to have another PR for it?

Comment (Piumal1999) [reply src/tutorial/modal.component.tsx:22]: > May be we can have like the content of the current walkthrough to the name of the walkthrough that's been clicked for now?

yeah. I think we can do it within this PR. (Demonstrating that _it loads the respective steps of the walkthrough when triggered_ would be enough for this PR )

**Comment by Vijaykv5** [comment]
Are we expecting like this? @Piumal1999 

https://github.com/openmrs/openmrs-esm-user-onboarding/assets/94985341/321092d6-67e0-4c1b-b143-795d71ac5cef

**Review by jayasanka-sack — APPROVED** [review]

**Pull request merged**